core/src/consensus/stake/mod.rs

```diff
@@ -322,25 +325,57 @@ pub fn get_validators(state: &TopLevelState) -> StateResult<Validators> {
 }
 
 pub fn add_intermediate_rewards(state: &mut TopLevelState, address: Address, reward: u64) -> StateResult<()> {
-    let mut rewards = IntermediateRewards::load_from_state(state)?;
-    rewards.add_quantity(address, reward);
-    rewards.save_to_state(state)?;
+    let metadata = state.metadata()?.expect("The metadata must exist");
+    let era = metadata.term_params().map_or(0, |p| p.era());
+    match era {
+        0 => {
+            let mut rewards = IntermediateRewardsV0::load_from_state(state)?;
+            rewards.add_quantity(address, reward);
+            rewards.save_to_state(state)?;
+        }
+        1 => {
+            let mut rewards = IntermediateRewardsV1::load_from_state(state)?;
+            rewards.add_quantity(address, reward);
+            rewards.save_to_state(state)?;
+        }
+        _ => unimplemented!(),
+    }
     Ok(())
 }
 
 pub fn drain_previous_rewards(state: &mut TopLevelState) -> StateResult<BTreeMap<Address, u64>> {
```

Comment:
I think it is more clear to the outside if it is namespaced.
There would be some approaches.
We can make them private and publicly reexport in the versioned mod.
```rust
fn add_intermediate_rewards;
fn drain_previous_rewards;
fn drain_current_rewards;

pub mod v0 {
	pub use super::add_intermediate_rewards;
	pub use super::drain_previous_rewards;
	...
}

pub mod v1 {
	pub use super::add_intermediate_rewards;
	pub use super::drain_current_rewards
	...
}
```
The other approach can be C++ style inline namespace style.
https://foonathan.net/2018/11/inline-namespaces/

```rust
pub mod v0 {
	pub fn add_intermediate_rewards;
	pub fn drain_previous_rewards;
	...
}

pub mod v1 {
	pub fn add_intermediate_rewards;
	pub fn drain_current_rewards
	...
}

pub fn add_intermediate_rewards(...) {
	if era == 0 {
		v0::add_intermediate_rewards(...);
	} else {
		v1::add_intermediate_rewards(...);
	}
}
pub use v1::{drain_current_rewards, ...};
```

Reply:
Fixed.